controllers/nodehealthcheck_controller.go

```diff
@@ -226,50 +229,38 @@ func (r *NodeHealthCheckReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return result, nil
 	}
 
-	template, err := resourceManager.GetTemplate(nhc.Spec.RemediationTemplate)
-	if err != nil {
-		log.Error(err, "failed to get template")
-		return result, err
-	}
-
 	// from here on these can change
 	patchInFlightRemediations = true
 
 	// delete remediation CRs for healthy nodes
 	for _, node := range healthyNodes {
-		remediationCR := resourceManager.GenerateRemediationCRBaseNamed(template.GroupVersionKind(), template.GetNamespace(), node.GetName())
-		if deleted, err := resourceManager.DeleteRemediationCR(remediationCR, nhc); err != nil {
-			log.Error(err, "failed to delete remediation CR for healthy node", "node", node.Name)
-			return result, err
-		} else if deleted {
-			log.Info("deleted remediation CR", "name", remediationCR.GetName(), "NHC name", nhc.Name)
-			r.Recorder.Eventf(nhc, eventTypeNormal, eventReasonRemediationRemoved, "Deleted remediation CR for node %s", remediationCR.GetName())
+		remediationCRs := resourceManager.GenerateAllRemediationCRs(&node, nhc)
+		for _, remediationCR := range remediationCRs {
+			if deleted, err := resourceManager.DeleteRemediationCR(remediationCR, nhc); err != nil {
+				log.Error(err, "failed to delete remediation CR for healthy node", "node", node.Name)
+				return result, err
```

Comment:
I might be missing something , but I don't think that we should block the deletion of the rest of the CRs in case the node is healthy

Reply:
I think when deletion of one CR failed, there is a good chance that others fail as well. So I think it's good enough to retry deletion of all CRs on next reconcile, which is triggered by returning the error. WDYT?

Reply:
Hmm, I didn't consider that.
Indeed it'll solve the issue in case there is a temporary issue which affect the deletion in general (temporary connectivity issue for example).
However it'll not solve it in case deletion problem is relevant to a specific CR (because in the next reconcile we will just try to delete the same problematic CR).
I still think a better solution would be not to block, but because of the first use-case I haven't thought about I consider this a Nit.

Reply:
> deletion problem is relevant to a specific CR

I'm not aware of a potential problem with a specific CR, which could cause a deletion failure permanently... 🤔 So IMHO the chances for a remporary issue with the API server, affecting all CRs, is much higher, and that's handled better by giving up for all CRs and retry later.

Reply:
existence of a finalizer and the owning operator is in a crash loop would do it

Reply:
but in that case the deletion timestamp would be set, which is handled in the delete method, and doesn't result in an error